### PR TITLE
Abstract background process with min delay

### PR DESCRIPTION
**Pull request by ujjwal-cyph**

### CODE REVIEW

**Comment by amanCypherock** [comment]
If the problem is with text not being readable, instead of adding a constant delay, we can add a glyph screen for transitions

File: src/level_four/core/tasks/receive_transaction_tasks.c [hunk 1]
```diff
@@ -133,7 +133,8 @@ void receive_transaction_tasks()
     } break;
 
     case RECV_TXN_DERIVE_ADD_SCREEN: {
-        instruction_scr_init(ui_text_deriving_address, NULL);
+        instruction_scr_init(ui_text_processing, NULL);
+        BSP_DelayMs(DELAY_TIME);
```
Comment (ujjwal-cyph) [inline src/level_four/core/tasks/receive_transaction_tasks.c:137]:
```suggestion
        delay_scr_init(ui_text_processing, DELAY_TIME);
```

File: src/level_four/core/tasks/receive_transaction_tasks_eth.c [hunk 2]
```diff
@@ -131,7 +131,8 @@ void receive_transaction_tasks_eth()
     } break;
 
     case RECV_TXN_DERIVE_ADD_SCREEN_ETH: {
-        instruction_scr_init(ui_text_deriving_address, NULL);
+        instruction_scr_init(ui_text_processing, NULL);
+        BSP_DelayMs(DELAY_TIME);
```
Comment (ujjwal-cyph) [inline src/level_four/core/tasks/receive_transaction_tasks_eth.c:135]:
```suggestion
        delay_scr_init(ui_text_processing, DELAY_TIME);
```

File: src/level_four/core/tasks/receive_transaction_tasks_near.c [hunk 3]
```diff
@@ -128,7 +128,8 @@ void receive_transaction_tasks_near() {
     } break;
 
     case RECV_TXN_DERIVE_ADD_SCREEN_NEAR: {
-      instruction_scr_init(ui_text_deriving_address, NULL);
+      instruction_scr_init(ui_text_processing, NULL);
+      BSP_DelayMs(DELAY_TIME);
```
Comment (ujjwal-cyph) [inline src/level_four/core/tasks/receive_transaction_tasks_near.c:132]:
```suggestion
        delay_scr_init(ui_text_processing, DELAY_TIME);
```

File: src/level_four/core/tasks/send_transaction_tasks.c [hunk 4]
```diff
@@ -83,6 +83,7 @@ void send_transaction_tasks()
 
     case SEND_TXN_VERIFY_COIN: {
         instruction_scr_init(ui_text_processing, NULL);
+        BSP_DelayMs(DELAY_TIME);
```
Comment (ujjwal-cyph) [inline src/level_four/core/tasks/send_transaction_tasks.c:86]:
```suggestion
        delay_scr_init(ui_text_processing, DELAY_TIME);
```

File: src/level_four/core/tasks/send_transaction_tasks.c [hunk 5]
```diff
@@ -199,7 +200,8 @@ void send_transaction_tasks()
     } break;
 
     case SEND_TXN_TAP_CARD_SEND_CMD: {
-        instruction_scr_init(ui_text_signing_transaction, NULL);
+        instruction_scr_init(ui_text_processing, NULL);
+        BSP_DelayMs(DELAY_TIME);
```
Comment (ujjwal-cyph) [inline src/level_four/core/tasks/send_transaction_tasks.c:204]:
```suggestion
        delay_scr_init(ui_text_processing, DELAY_TIME);
```

File: src/level_four/core/tasks/send_transaction_tasks_eth.c [hunk 6]
```diff
@@ -87,6 +87,7 @@ void send_transaction_tasks_eth()
 
     case SEND_TXN_VERIFY_COIN_ETH: {
         instruction_scr_init(ui_text_processing, NULL);
+        BSP_DelayMs(DELAY_TIME);
```
Comment (ujjwal-cyph) [inline src/level_four/core/tasks/send_transaction_tasks_eth.c:90]:
```suggestion
        delay_scr_init(ui_text_processing, DELAY_TIME);
```

File: src/level_four/core/tasks/send_transaction_tasks_eth.c [hunk 7]
```diff
@@ -154,7 +155,8 @@ void send_transaction_tasks_eth()
     } break;
 
     case SEND_TXN_CALCULATE_AMOUNT_ETH: {
-        instruction_scr_init(ui_text_fetching_recipient_amount, NULL);
+        instruction_scr_init(ui_text_processing, NULL);
+        BSP_DelayMs(DELAY_TIME);
```
Comment (ujjwal-cyph) [inline src/level_four/core/tasks/send_transaction_tasks_eth.c:159]:
```suggestion
        delay_scr_init(ui_text_processing, DELAY_TIME);
```

File: src/level_four/core/tasks/send_transaction_tasks_eth.c [hunk 8]
```diff
@@ -273,7 +275,8 @@ void send_transaction_tasks_eth()
     } break;
 
     case SEND_TXN_TAP_CARD_SEND_CMD_ETH: {
-        instruction_scr_init(ui_text_signing_transaction, NULL);
+        instruction_scr_init(ui_text_processing, NULL);
+        BSP_DelayMs(DELAY_TIME);
```
Comment (ujjwal-cyph) [inline src/level_four/core/tasks/send_transaction_tasks_eth.c:279]:
```suggestion
        delay_scr_init(ui_text_processing, DELAY_TIME);
```

File: src/level_four/core/tasks/send_transaction_tasks_near.c [hunk 9]
```diff
@@ -78,6 +78,7 @@ void send_transaction_tasks_near() {
 
     case SEND_TXN_VERIFY_COIN_NEAR: {
         instruction_scr_init(ui_text_processing, NULL);
+        BSP_DelayMs(DELAY_TIME);
```
Comment (ujjwal-cyph) [inline src/level_four/core/tasks/send_transaction_tasks_near.c:81]:
```suggestion
        delay_scr_init(ui_text_processing, DELAY_TIME);
```

File: src/level_four/core/tasks/send_transaction_tasks_near.c [hunk 10]
```diff
@@ -134,7 +135,8 @@ void send_transaction_tasks_near() {
     } break;
 
     case SEND_TXN_CALCULATE_AMOUNT_NEAR: {
-        instruction_scr_init(ui_text_fetching_recipient_amount, NULL);
+        instruction_scr_init(ui_text_processing, NULL);
+        BSP_DelayMs(DELAY_TIME);
```
Comment (ujjwal-cyph) [inline src/level_four/core/tasks/send_transaction_tasks_near.c:139]:
```suggestion
        delay_scr_init(ui_text_processing, DELAY_TIME);
```

File: src/level_four/core/tasks/send_transaction_tasks_near.c [hunk 11]
```diff
@@ -224,7 +226,8 @@ void send_transaction_tasks_near() {
     } break;
 
     case SEND_TXN_TAP_CARD_SEND_CMD_NEAR: {
-        instruction_scr_init(ui_text_signing_transaction, NULL);
+        instruction_scr_init(ui_text_processing, NULL);
+        BSP_DelayMs(DELAY_TIME);
```
Comment (ujjwal-cyph) [inline src/level_four/core/tasks/send_transaction_tasks_near.c:230]:
```suggestion
        delay_scr_init(ui_text_processing, DELAY_TIME);
```

**Comment by ujjwal-cyph** [comment]
Tested Add coin (1 and multiple), Receive (Ethereum, Dash, Near), Send (Ethereum)

**Pull request merged**